pkg/collector/corechecks/oracle-dbm/oracle.go

```diff
@@ -200,9 +217,38 @@ func (c *Check) Connect() (*sqlx.DB, error) {
 		}
 	}
 
+	if c.config.AgentSQLTrace.Enabled {
+		db.SetMaxOpenConns(1)
+		_, err := db.Exec("ALTER SESSION SET tracefile_identifier='DDAGENT'")
+		if err != nil {
+			log.Warnf("failed to set tracefile_identifier: %v", err)
+		}
+
+		/* We are concatenating values instead of passing parameters, because there seems to be a problem
+		 * in go-ora with passing bool parameters to PL/SQL. As a mitigation, we are asserting that the
+		 * parameters are bool
+		 */
+		binds := returnFalseIfNotBool(c.config.AgentSQLTrace.Binds)
+		waits := returnFalseIfNotBool(c.config.AgentSQLTrace.Waits)
+		setEventsStatement := fmt.Sprintf("BEGIN dbms_monitor.session_trace_enable (binds => %t, waits => %t); END;", binds, waits)
+		log.Trace("trace statement: %s", setEventsStatement)
+		_, err = db.Exec(setEventsStatement)
+		if err != nil {
+			log.Errorf("failed to set SQL trace: %v", err)
+		}
+		if c.config.AgentSQLTrace.TracedRuns == 0 {
+			c.config.AgentSQLTrace.TracedRuns = DEFAULT_SQL_TRACED_RUNS
+		}
+	}
+
 	return db, nil
 }
 
+func returnFalseIfNotBool(input interface{}) bool {
+	retValue, _ := input.(bool)
+	return retValue
+}
```

Comment:
This really looks weird to me because we're adding `interface{}` to make a type check that the compiler can already guarantee otherwise. Why not make this just keep the types and ensure that thing do fail at compilation if someone was to accidentally change the type of `AgentSQLTrace.Binds`? 

```suggestion
func isTrue(val bool) bool {
	return val
}
```

So now if someone changes Binds to a string, this won't compile instead of having a `false` inferred from the wrong type at runtime.